src/osd/PGLog.cc

```diff
@@ -1006,14 +1007,34 @@ void PGLog::read_log(ObjectStore *store, coll_t pg_coll,
 	bv);
       if (r >= 0) {
 	object_info_t oi(bv);
+
 	if (oi.version < i->version) {
 	  ldpp_dout(dpp, 15) << "read_log  missing " << *i
-			     << " (have " << oi.version << ")" << dendl;
+			     << " (have " << oi.version << ")"
+			     << " unmodified_omap " << i->unmodified_omap
+			     << " unmodified_extents " << i->unmodified_extents << dendl;
 	  missing.add(i->soid, i->version, oi.version);
+          missing.missing[i->soid].unmodified_omap = i->unmodified_omap;
+          missing.missing[i->soid].unmodified_extents.insert(i->unmodified_extents);
+          list<pg_log_entry_t>::const_iterator piter = log.log.end();
+          map<hobject_t, pg_missing_t::item, hobject_t::ComparatorWithDefault>::iterator miter
+            = missing.missing.find(i->soid);
+          if (i->prior_version >= log.tail)
+            piter = log.find_entry(i->prior_version);
+          while(piter != log.log.end()) {
+            miter->second.unmodified_omap = miter->second.unmodified_omap && piter->unmodified_omap;
+            miter->second.unmodified_extents.intersection_of(piter->unmodified_extents);
+            eversion_t prior_version = piter->prior_version;
+            piter = log.find_entry(prior_version);
+          }
 	}
       } else {
-	ldpp_dout(dpp, 15) << "read_log  missing " << *i << dendl;
+	ldpp_dout(dpp, 15) << "read_log  missing " << *i 
+			   << " unmodified_omap " << i->unmodified_omap
+			   << " unmodified_extents " << i->unmodified_extents << dendl;
 	missing.add(i->soid, i->version, eversion_t());
```

Comment:
No, here, we are saying that we *no copy of the object*.  You have to initialize the unmodified information to conservatively recover the whole object.